test/hw_tests/hwtest_timestamp_standalone.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2021 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+#include <numeric>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "psen_scan_v2_standalone/core.h"
+
+#include "psen_scan_v2/test_data.h"
+#include "psen_scan_v2/test_data_assembler.h"
+
+using namespace psen_scan_v2_standalone;
+
+namespace psen_scan_v2_test
+{
+static const std::string SCANNER_IP{ "192.168.0.10" };
+static const int HOST_UDP_DATA_PORT{ 55007 };
+static const util::TenthOfDegree ANGLE_START{ 1 };
+static const util::TenthOfDegree ANGLE_END{ 2749 };
+
+static constexpr int64_t SCANNER_RUN_DURATION_S{ 30 };
+static constexpr std::size_t MINIMUM_TEST_SIZE{ 900 };
+static constexpr int64_t TIME_COMPARISON_EPSILON_NS{ 1000000 };
+static const std::string UDP_DATA_FILENAME_ENV_VAR{ "UDP_DATA_FILENAME" };
+
+class TimestampTests : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    const char* udp_data_filename{ std::getenv(UDP_DATA_FILENAME_ENV_VAR.c_str()) };
+    ASSERT_NE(udp_data_filename, nullptr)
+        << "Searching environment variable " << UDP_DATA_FILENAME_ENV_VAR << "failed.";
+    udp_data_filename_ = udp_data_filename;
+
+    buildScannerConfig();
+    setLogLevel(CONSOLE_BRIDGE_LOG_INFO);
+    ASSERT_GE(testSize(), MINIMUM_TEST_SIZE) << "Assembling test data failed.";
+  }
+
+protected:
+  const TestData& testData() const
+  {
+    static const std::unique_ptr<TestData> test_data_ptr{ TestDataAssembler::assemble(
+        *scanner_config_, SCANNER_RUN_DURATION_S, udp_data_filename_, HOST_UDP_DATA_PORT) };
+    return *test_data_ptr;
```

Comment:
The static is there to prevent having to gather the data again? Why don't you use a member instead?
And I think you might be creating copies here.